extensions/binding/org.eclipse.smarthome.binding.fsinternetradio/src/main/java/org/eclipse/smarthome/binding/fsinternetradio/internal/FSInternetRadioHandlerFactory.java

```diff
@@ -22,16 +22,24 @@
 import org.eclipse.smarthome.core.thing.ThingTypeUID;
 import org.eclipse.smarthome.core.thing.binding.BaseThingHandlerFactory;
 import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandlerFactory;
+import org.eclipse.smarthome.io.net.http.CommonHttpClient;
+import org.eclipse.smarthome.io.net.http.HttpClientFactory;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.ConfigurationPolicy;
+import org.osgi.service.component.annotations.Reference;
 
 /**
  * The {@link FSInternetRadioHandlerFactory} is responsible for creating things and thing
  * handlers.
  *
  * @author Patrick Koenemann - Initial contribution
  */
+@Component(service = ThingHandlerFactory.class, immediate = true, configurationPid = "binding.fsinternetradio", configurationPolicy = ConfigurationPolicy.OPTIONAL)
```

Comment:
Hm, IIRC we don't need `immediate` for the handler factory. If no one is interested in thing handler factories, no need to have it. But there should be in every ESH instance "someone" that collects the references...

Reply:
I did that change in line with [the code that is generated by our archetype](https://github.com/eclipse/smarthome/blob/master/tools/archetype/binding/src/main/resources/archetype-resources/src/main/java/internal/__bindingIdCamelCase__HandlerFactory.java#L43).

Reply:
If you prefer to discuss it in a separate issue and then fixing this class (if necessary later), see #5163
I hoped we could fix this on the go...